systemvm/patches/debian/config/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -95,9 +95,25 @@ def get_control_if(self):
                 return ip
         return None
 
+    def check_if_link_exists(self,dev):
+        cmd="ip link show dev %s"%dev
+        result=CsHelper.execute(cmd)
+        if(len(result)!=0):
+           return True
+        else:
+           return False
+
+    def check_if_link_up(self,dev):
+        cmd="ip link show dev %s | tr '\n' ' ' | cut -d ' ' -f 9"%dev
```

Comment:
We should keep as much logic in Python as possible. The performance difference is small, but with tr and grep you spawn subprocessess again.

But as @remibergsma says. Try to open that file in /sys and parse the contents. You can use the simple Python file functions.

No need to execute IP. The less subprocesses we execute, the better.